src/wx/opts.cpp

```diff
@@ -135,18 +135,22 @@ const wxString joynames[NUM_KEYS] = {
     wxT("Speed"), wxT("Capture"), wxT("GS")
 };
 
-wxJoyKeyBinding defkeys[NUM_KEYS * 2] = {
-    WJKB(WXK_UP), WJKB(1, WXJB_AXIS_MINUS, 1), WJKB(WXK_DOWN), WJKB(1, WXJB_AXIS_PLUS, 1),
-    WJKB(WXK_LEFT), WJKB(0, WXJB_AXIS_MINUS, 1), WJKB(WXK_RIGHT), WJKB(0, WXJB_AXIS_PLUS, 1),
-    WJKB(wxT('X')), WJKB(0, WXJB_BUTTON, 1), WJKB(wxT('Z')), WJKB(1, WXJB_BUTTON, 1),
-    WJKB(wxT('A')), WJKB(2, WXJB_BUTTON, 1), WJKB(wxT('S')), WJKB(3, WXJB_BUTTON, 1),
-    WJKB(WXK_BACK), WJKB(4, WXJB_BUTTON, 1), WJKB(WXK_RETURN), WJKB(5, WXJB_BUTTON, 1),
-    WJKB(WXK_NUMPAD_UP), WJKB(0), WJKB(WXK_NUMPAD_DOWN), WJKB(0),
-    WJKB(WXK_NUMPAD_LEFT), WJKB(0), WJKB(WXK_NUMPAD_RIGHT), WJKB(0),
-    WJKB(WXK_NUMPAD_PAGEUP), WJKB(0), WJKB(WXK_NUMPAD_PAGEDOWN), WJKB(0),
-    WJKB(wxT('W')), WJKB(0), WJKB(wxT('Q')), WJKB(0),
-    WJKB(WXK_SPACE), WJKB(0), WJKB(0), WJKB(0),
-    WJKB(0), WJKB(0)
+wxJoyKeyBinding defkeys_keyboard[NUM_KEYS] = {
+    WJKB(WXK_UP), WJKB(WXK_DOWN), WJKB(WXK_LEFT), WJKB(WXK_RIGHT),
+    WJKB(wxT('Z')), WJKB(wxT('X')), WJKB(wxT('A')), WJKB(wxT('S')),
+    WJKB(wxT('C')), WJKB(wxT('V')),
+    WJKB(0), WJKB(0), WJKB(0), WJKB(0),
+    WJKB(0), WJKB(0), WJKB(0), WJKB(0),
+    WJKB(WXK_SPACE), WJKB(0), WJKB(0)
+};
+
+wxJoyKeyBinding defkeys_joystick[NUM_KEYS] = {
+    WJKB(1, WXJB_AXIS_MINUS, 1), WJKB(1, WXJB_AXIS_PLUS, 1), WJKB(0, WXJB_AXIS_MINUS, 1), WJKB(0, WXJB_AXIS_PLUS, 1),
+    WJKB(0, WXJB_BUTTON, 1), WJKB(1, WXJB_BUTTON, 1), WJKB(2, WXJB_BUTTON, 1), WJKB(3, WXJB_BUTTON, 1),
+    WJKB(4, WXJB_BUTTON, 1), WJKB(5, WXJB_BUTTON, 1),
+    WJKB(0), WJKB(0), WJKB(0), WJKB(0),
+    WJKB(0), WJKB(0), WJKB(0), WJKB(0),
+    WJKB(0), WJKB(0), WJKB(0)
```

Comment:
I forgot to mention, but this removed some mappings. Basically, we keep the binding only for the standard buttons (the one you have on an actual GBA). Everything else was set to empty.

Reply:
Is this the mapping for the xbox 360 controller? Removing the unnecessary bindings is good.

Reply:
This is the same from before for controller. I don't have one to check if these match.